src/box/lua/net_box.lua

```diff
@@ -826,9 +826,16 @@ local function stream_new_stream(stream)
     return stream._conn:new_stream()
 end
 
-local function stream_begin(stream, opts)
+local function stream_begin(stream, args, opts)
     check_remote_arg(stream, 'begin')
-    local res = stream:_request(M_BEGIN, opts, nil, stream._stream_id)
+    if args ~= nil and type(args) ~= 'table' then
+        error("Use remote:begin({timeout})")
+    end
+    local timeout = args and args.timeout
+    if timeout and (type(timeout) ~= "number" or timeout <= 0) then
+        error("timeout must be a number greater than 0")
+    end
```

Comment:
I didn't find a place in the code where we check that a timeout must be positive. Can we just handle negative timeouts here and in box.begin()?

BTW a zero timeout should be fine - it means no yields in a transaction.

Reply:
Ok, fixed, also add asserts in txn_set_timeout to check that timeout is always positive

Reply:
Stream transactions always yields! Because we send next message from iproto to tx only after processing previous one.